tests/test_brownian_interval.py

```diff
@@ -0,0 +1,311 @@
+            val = (val,)
+        cache[ta_, tb_] = tuple(v.detach().clone() for v in val)
+
+    cache2 = {}
+    for ta_, tb_ in cache:
+        val = bm(ta_, tb_, return_U=return_U, return_A=return_A)
+        if torch.is_tensor(val):
+            val = (val,)
+        cache2[ta_, tb_] = tuple(v.detach().clone() for v in val)
+
+    for ta_, tb_ in cache:
+        for v1, v2 in zip(cache[ta_, tb_], cache2[ta_, tb_]):
+            assert (v1 == v2).all()
+
+
+@pytest.mark.parametrize("device", devices)
+@pytest.mark.parametrize("levy_area_approximation", ['none', 'space-time', 'davie', 'foster'])
+def test_normality_simple(device, levy_area_approximation):
+    if device == gpu and not torch.cuda.is_available():
+        pytest.skip(msg="CUDA not available.")
+
+    t0_, t1_ = 0.0, 1.0
+    t0, t1 = torch.tensor([t0_, t1_], device=device)
+    for _ in range(REPS):
+        W = torch.tensor(npr.randn(), device=device).repeat(LARGE_BATCH_SIZE)
+        bm = torchsde.BrownianInterval(t0=t0, t1=t1, W=W, levy_area_approximation=levy_area_approximation)
+
+        for _ in range(REPS):
+            t_ = npr.uniform(low=t0_, high=t1_)
+            samples = bm(t_)
+            samples_ = samples.cpu().detach().numpy()
+
+            mean_ = W.cpu() * (t_ - t0_) / (t1_ - t0_)
+            std_ = np.sqrt((t1_ - t_) * (t_ - t0_) / (t1_ - t0_))
+            ref_dist = norm(loc=mean_, scale=std_)
+
+            _, pval = kstest(samples_, ref_dist.cdf)
+            assert pval >= ALPHA
+
+
+@pytest.mark.parametrize("device", devices)
+@pytest.mark.parametrize("levy_area_approximation", ['none', 'space-time', 'davie', 'foster'])
+def test_normality_conditional(device, levy_area_approximation):
+    if device == gpu and not torch.cuda.is_available():
+        pytest.skip(msg="CUDA not available.")
+
+    t0_, t1_ = 0.0, 1.0
+    t0, t1 = torch.tensor([t0_, t1_], device=device)
+    for _ in range(REPS):
+        bm = torchsde.BrownianInterval(t0=t0, t1=t1, shape=(LARGE_BATCH_SIZE,),
+                                       levy_area_approximation=levy_area_approximation)
+
+        for _ in range(REPS):
+            t_ = npr.uniform(low=t0_, high=t1_)
+            ta = npr.uniform(low=t0_, high=t1_)
+            tb = npr.uniform(low=t0_, high=t1_)
+            ta, t_, tb = sorted([t_, ta, tb])
+
+            increment = bm(ta, tb).cpu().detach().numpy()
+            sample = bm(ta, t_).cpu().detach().numpy()
+
+            mean = increment * (t_ - ta) / (tb - ta)
+            std = np.sqrt((tb - t_) * (t_ - ta) / (tb - ta))
+            rescaled_sample = (sample - mean) / std
+
+            _, pval = kstest(rescaled_sample, 'norm')
+            assert pval >= ALPHA
+
+
+@pytest.mark.parametrize("device", devices)
+@pytest.mark.parametrize("levy_area_approximation", ['none', 'space-time', 'davie', 'foster'])
+@pytest.mark.parametrize("random_order", [False, True])
+def test_continuity(device, levy_area_approximation, random_order):
```

Comment:
What is the main purpose of this test?

Reply:
As the name suggests, to test continuity ;)

I found (pre-bugfixes) that `BrownianInterval` would sometimes produce discontinuous paths, so this felt like a sensible sanity check for that issue.

Reply:
That's interesting, do you have any guesses as to why it would produce discontinuous paths?